src/osd/OSDMap.cc

```diff
@@ -1481,6 +1540,19 @@ int OSDMap::apply_incremental(const Incremental &inc)
       (*primary_temp)[p->first] = p->second;
   }
 
+  for (auto& p : inc.new_pg_remap) {
```

Comment:
just looking at the data structure removing the old entries before adding the new ones looks fragile, since a new remap of the same pg would be erased if we were erasing an old remap of the same pg. I see later commits only use addition or removal of a remap for a given pg though, so this isn't a bug as is